src/nitypes/scalar.py

```diff
@@ -217,6 +290,9 @@ def __str__(self) -> str:
         return value_str
 
     def _check_units_equal_for_comparison(self, other_units: str) -> None:
-        """Raise a ValueError if other_units != self.units."""
+        """Raise a ValueError if self.units != other_units."""
         if self.units != other_units:
             raise ValueError("Comparing Scalar objects with different units is not permitted.")
+
+    def _comparing_numeric_and_string_not_permitted(self) -> TypeError:
+        return TypeError("Comparing Scalar objects of numeric and string types is not permitted.")
```

Comment:
This does not need to be a member of the class. Let's make it a free function, like the functions in `_exceptions.py`.

```suggestion
def _comparing_numeric_and_string_not_permitted() -> TypeError:
    return TypeError("Comparing Scalar objects of numeric and string types is not permitted.")
```

Also replace `raise self._comparing_numeric_and_string_not_permitted()` with `raise _comparing_numeric_and_string_not_permitted()`

Reply:
I will make it so.